external/mmdetection/detection_tasks/apis/detection/inference_task.py

```diff
@@ -218,15 +218,18 @@ def _add_predictions_to_dataset(self, prediction_results, dataset, confidence_th
                             polygon = Polygon(points=points)
                             if polygon.get_area() > 1e-12:
                                 shapes.append(Annotation(polygon, labels=labels, id=ID(f"{label_idx:08}")))
+                    if features:
+                        features = [features,
+                                    draw_instance_segm_saliency_map(shapes, dataset_item, self._labels)]
```

Comment:
So, if the `add_saliency_map` is False, draw_instance_segm_saliency_map is still draws a saliency map?

Reply:
You're such a great code reviewer!